### ISSUE TITLE
Forbid implicit `enumerate` calls

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#1224]
# Rule request

## Thesis

This code should raise:

```python
x = 0
for i in items:
    x += 1
    print(items[x])
```

But, it is not! 

Why is it a bad idea? We are using some mutable state from outside world, when we can just use `enumerate`.

How to detect this pattern?

- variable is assigned as `0`
- variable then is used on `for` iterable as index
- variable is adjusted by `1` inside the loop's body

Related https://github.com/wemake-services/wemake-python-styleguide/issues/689

This is refactoring pattern. Looks rather complex to me.

**Comment by sobolevn** [comment]
Handled by `ruff` and `SIM113`
